src/Components/Explore/Explore.jsx

```diff
@@ -0,0 +1,83 @@
+import React, { useState, useEffect } from 'react';
+import Post from '../Post/Post';
+import { FiFilter } from 'react-icons/fi';
+import { FaHeart } from 'react-icons/fa';
+import { IoIosArrowForward } from 'react-icons/io';
+import { GiSandsOfTime } from 'react-icons/gi';
+import { BiComment } from 'react-icons/bi';
+import { useSelector } from 'react-redux';
+import Spinner from '../Spinner/Spinner';
+import './Explore.css';
+
+const Explore = () => {
+  const { posts, loading } = useSelector((state) => state.post);
+  const [sortBy, setSortBy] = useState('latest');
+  const [sortedPosts, setSortedPosts] = useState([...posts]);
+
+  useEffect(() => {
+    setSortedPosts([...posts]);
+  }, [posts]);
+
+  if (sortBy === 'liked') {
```

Comment:
suggestion - I think any changes made to state variable should be done through the setState, sort is an impure function. but for now it does not require re-render so its working.